src/python/sortingview/gui/extensions/timeseries/TimeseriesViewNew/TimeseriesWidgetNew.tsx

```diff
@@ -186,33 +194,61 @@ const TimeseriesWidgetNew = (props: Props) => {
         }
         setPanels(panels0)
     }, [channel_ids, setPanels, timeseriesData, y_scale_factor, visibleChannelIds, width, recordingSelection.ampScaleFactor, recordingSelection.timeRange])
-    useEffect(() => {
-        if (actions === null) {
-            const a: TimeWidgetAction[] = [
-                {
-                    type: 'button',
-                    callback: _handleScaleAmplitudeUp,
-                    title: 'Scale amplitude up [up arrow]',
-                    icon: <FaArrowUp />,
-                    keyCode: 38
-                },
-                {
-                    type: 'button',
-                    callback: _handleScaleAmplitudeDown,
-                    title: 'Scale amplitude down [down arrow]',
-                    icon: <FaArrowDown />,
-                    keyCode: 40
-                },
-                {
-                    type: 'divider'
-                }
-            ]
-            setActions(a)
+    const actions = useMemo(() => {
```

Comment:
The `useMemo` dependency on the set of actions means you don't need the `useEffect` hook, which already had a kind-of-hacky run-once measure added! That's much more elegant than what I'd done before (at least I think this part was my code...)

Reply:
> The `useMemo` dependency on the set of actions means you don't need the `useEffect` hook, which already had a kind-of-hacky run-once measure added! That's much more elegant than what I'd done before (at least I think this part was my code...)

I used useEffect 'way back' when I didn't really understand how to use useMemo(). :)